src/components/Table/Table.tsx

```diff
@@ -1,13 +1,13 @@
 import clsx from 'clsx';
-import React, { FC, useEffect, useMemo } from 'react';
+import React, { useEffect, useMemo } from 'react';
 import { HeaderGroup, usePagination, useSortBy, useTable } from 'react-table';
 
 import { bemCls, getVariantClasses } from '../../utils';
 
 import { defaultSortingIcons } from './Table.const';
-import { TableProps } from './Table.types';
+import { TableData, TableProps } from './Table.types';
 
-const Table: FC<TableProps<object>> = ({
+const Table = <D extends TableData>({
```

Comment:
Why don't you use the more standard T instead of D for the generic?

Reply:
D stands for Data, you can name those whatever you want. The T you see in most of the examples just stands for Type. In the new docs they're now using `Type`: https://www.typescriptlang.org/docs/handbook/2/generics.html